crates/bevy_ecs/src/system/query.rs

```diff
@@ -292,6 +292,31 @@ where
         }
     }
 
+    /// Gets the result of a single-result query
+    ///
+    /// If the query has exactly one result, returns the result inside `Ok`
+    /// otherwise returns either `Err(QuerySingleError::NoEntities(...))`
+    /// or `Err(QuerySingleError::MultipleEntities(...))`
+    ///
+    /// # Examples
+    ///
+    /// ```
+    ///  # use bevy_ecs::system::{Query, QuerySingleError};
+    ///  # use bevy_render::camera::Camera;
```

Comment:
```suggestion
    ///  # struct Camera;
```

Reply:
That being said, I don't know if `Camera` is necessarily the best thing to be suggesting people use this API for.

For example, a normal game will have a 2d or 3d camera and a UI camera.

Maybe 
```rust
/// Marks a player which is controlled by the keyboard on this machine.
struct LocalKeyboardControlled;
```
Although again, that's not always ideal. But this is more arguing about the merits of this API in general - ideally this example would move something out of a variable, so that it's clear that it can only be used once.

Reply:
That's a great idea! thanks